Neo4j.Driver/Neo4j.Driver/Internal/IO/ChunkReader.cs

```diff
@@ -41,33 +40,26 @@ public ChunkReader(Stream downStream)
             
         }
 
-        public ChunkReader(Stream downStream, ILogger logger)
-            : this(downStream, new MemoryStream(), logger)
-        {
-            
-        }
-
-        internal ChunkReader(Stream downStream, MemoryStream chunkStream, ILogger logger)
+        internal ChunkReader(Stream downStream, ILogger logger)
         {
             Throw.ArgumentNullException.IfNull(downStream, nameof(downStream));
             Throw.ArgumentOutOfRangeException.IfFalse(downStream.CanRead, nameof(downStream));
 
-            Throw.ArgumentNullException.IfNull(chunkStream, nameof(chunkStream));
-
             _downStream = downStream;
-            _chunkStream = chunkStream;
             _logger = logger;
         }
 
         private bool TryReadFromBuffer(Stream targetStream)
         {
             while (true)
             {
+                // First try to retrieve the chunk size.
                 if (_currentChunkSize == -1 && HasBytesAvailable(_chunkSizeBuffer.Length))
                 {
-                    ReadFromChunkStream(_chunkSizeBuffer, 0, _chunkSizeBuffer.Length);
+                    ReadFromBuffer(_chunkSizeBuffer, 0, _chunkSizeBuffer.Length);
```

Comment:
This method is only used for reading chunk size, maybe we could just move all logic of reading size header into the same method and rename it as `ReadChunkSize()`?